src/plugins/intel_cpu/src/node.cpp

```diff
@@ -316,6 +315,103 @@ void Node::selectPreferPrimitiveDescriptor(const std::vector<impl_desc_type>& pr
     selectPrimitiveDescriptorByIndex(0);
 }
 
+bool Node::checkScalarShape(const ov::PartialShape& pshape) {
+    int oneNum = 0;
+    int sz = static_cast<int>(pshape.size());
+    for (auto s : pshape) {
+        if (s.is_static() && s.get_length() == 1) {
+            oneNum++;
+        }
+    }
+    return oneNum >= sz - 1;
+}
+
+void Node::selectPreferPrimitiveDescriptorWithShape(const std::vector<impl_desc_type>& priority, bool ignoreConstInputs) {
+    for (auto& type : priority) {
```

Comment:
Could you please extract logical pieces of declarative actions from this gigantic multibranch triple nesting 'for loop' and define them as named functions / lambdas?
We better to have it as something like:
``` c++
void Node::selectPreferPrimitiveDescriptorWithShape(const std::vector<impl_desc_type>& priority, bool ignoreConstInputs) {
    for (auto& type : priority) {
        auto something = getSomethingFrom(thing);
        doSomethingWith(something);
        ...
    }
}
```
Currently the logic of this function is really hard to follow

Reply:
Yes, it's hard to read, so I use lambda function replace loop.